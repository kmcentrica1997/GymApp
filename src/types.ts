export type NavigatorParamsList = {
  Home: undefined;
  DashBoard: undefined;
};
