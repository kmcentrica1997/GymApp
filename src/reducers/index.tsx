import {combineReducers} from 'redux';
import TodosReducer from './todos';

export default combineReducers({
  todos: TodosReducer,
});
