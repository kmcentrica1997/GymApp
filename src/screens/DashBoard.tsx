import {StackNavigationProp} from '@react-navigation/stack';
import React, {FC, useEffect} from 'react';
import {Text, StyleSheet, View, Image, FlatList} from 'react-native';
import {useDispatch, RootStateOrAny, useSelector} from 'react-redux';
import {Loading} from '../components/Loading/Loading';
import * as actions from '../redux/actions/users';
import {NavigatorParamsList} from '../types';

export interface DashBoardProps {
  navigation: StackNavigationProp<NavigatorParamsList, 'DashBoard'>;
}

const DashBoard: FC<DashBoardProps> = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(actions.getUsersRequest());
  }, [dispatch]);

  const allUsers = useSelector((state: RootStateOrAny) => state.users.items);

  const _renderItem = ({item}: any) => {
    console.log(item);
    return (
      <View>
        <Text style={styles.text} key={item.index}>
          {item.firstName}
        </Text>
        <Image
          style={styles.image2}
          source={require('/Users/km00776/Desktop/RNprojects/GymApp/src/img/gym.png')}
        />
      </View>
    );
  };

  // useEffect(() => {
  //   (async () => {
  //     console.log(
  //       getUsers()
  //         .then(res => {
  //           console.log(res.data.data);
  //         })
  //         .catch(err => console.log(err.response)),
  //     );
  //   })();
  // }, []);

  // perhaps set timer?

  return (
    <View style={styles.container}>
      {!allUsers?.length && <Loading />}
      {/* {allUsers &&
        allUsers.map((user: any) => {
          return (
            <Text key={user.id} style={styles.text}>
              <Image
                style={styles.image2}
                source={require('/Users/km00776/Desktop/RNprojects/GymApp/src/img/gym.png')}
              />
              {user.firstName}
            </Text>
          );
        })} */}
      <FlatList
        data={allUsers}
        renderItem={_renderItem}
        style={styles.list}
        keyExtractor={index => index.id}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignItems: 'center',
    marginTop: 30,
    marginLeft: 34,
    justifyContent: 'center',
    flex: 1,
  },
  list: {
    width: 100,
    flex: 1,
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
  text: {
    fontFamily: 'Cochin',
    fontSize: 20,
    color: 'black',
  },
  image: {
    marginLeft: 5,
    width: 100,
    height: 50,
  },

  image2: {
    marginLeft: 5,
    width: 50,
    height: 50,
  },
  loading: {
    textAlign: 'center',
    fontWeight: 'bold',
    marginTop: 300,
    width: 200,
    fontSize: 30,
    color: 'white',
  },
});
export default DashBoard;
