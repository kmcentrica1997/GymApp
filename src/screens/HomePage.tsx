import React, {FC} from 'react';
import {View, StyleSheet} from 'react-native';

const Home: FC = props => {
  return <View />;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
export default Home;
