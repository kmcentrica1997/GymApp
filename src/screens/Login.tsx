import {StackNavigationProp} from '@react-navigation/stack';
import React, {FC} from 'react';
import {View, Text, StyleSheet, Image} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {NavigatorParamsList} from '../types';
import {colors} from '../utils/colors';

export interface HomeProps {
  navigation: StackNavigationProp<NavigatorParamsList, 'Home'>;
}

const Login: FC<HomeProps> = ({navigation}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Hello!</Text>
      <Image
        style={styles.img}
        source={require('/Users/km00776/Desktop/RNprojects/GymApp/src/img/gateway.png')}
      />
      <TouchableOpacity
        testID="myBtn"
        style={styles.btn}
        onPress={() => navigation.navigate('DashBoard')}
        accessibilityLabel="Button to redirect to Dashboard">
        <Text>Login</Text>
      </TouchableOpacity>
    </View>
  );
};
export default Login;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 40,
    fontWeight: 'bold',
    marginBottom: 300,
    color: colors.darkBlue,
  },
  img: {
    width: 250,
    height: 250,
    justifyContent: 'center',
    bottom: 200,
  },
  btn: {
    backgroundColor: 'red',
  },
});
