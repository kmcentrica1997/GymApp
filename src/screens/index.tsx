import Login from './Login';
import Home from './HomePage';
import DashBoard from './DashBoard';

export {Login, Home, DashBoard};
