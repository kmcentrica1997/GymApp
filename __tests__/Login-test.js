import {Login} from '../src/screens';
import React from 'react';
import {create} from 'react-test-renderer';

const navigation = {
  navigate: jest.fn(),
};

const tree = create(<Login navigation={navigation} />);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

test('button pressed', () => {
  const button = tree.root.findByProps({testID: 'myBtn'}).props;
  button.onPress();

  expect(navigation.navigate).toBeCalledWith('DashBoard');
});
