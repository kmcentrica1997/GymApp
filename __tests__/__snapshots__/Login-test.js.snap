// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#252250",
        "fontSize": 40,
        "fontWeight": "bold",
        "marginBottom": 300,
      }
    }
  >
    Hello!
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/img/gateway.png",
      }
    }
    style={
      Object {
        "bottom": 200,
        "height": 250,
        "justifyContent": "center",
        "width": 250,
      }
    }
  />
  <View
    accessibilityLabel="Button to redirect to Dashboard"
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myBtn"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#252250",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
